ldk/javascript/examples/self-test-loop/src/tests/filesystem/index.ts

```diff
@@ -293,3 +293,76 @@ export const testFileExists = (): Promise<boolean> =>
         });
     });
   });
+
+export const testOpenFile = (): Promise<boolean> =>
+  new Promise(async (resolve, reject) => {
+    const filePath = `test.txt`;
+    const writeMode = 0o755;
+
+    network
+      .encode('some text')
+      .then((encodedValue) => {
+        setTimeout(() => {
+          filesystem
+            .writeFile({
+              path: filePath,
+              data: encodedValue,
+              writeOperation: filesystem.WriteOperation.overwrite,
+              writeMode,
+            })
+            .then(() => {
```

Comment:
I'm actually rebuilding filesystem test group in await async pattern right now. Perhaps it would makes sense to drop the act().then().catch() pattern here to be aligned.